operator/main.py

```diff
@@ -464,22 +466,203 @@ def handle_modified(core_v1_client, obj):
+
+    # For each brick, add brick path and node id
+    bricks = obj["spec"]["storage"]
+    for idx, storage in enumerate(bricks):
+        data["bricks"].append({
+            "brick_path": "/bricks/%s/data/brick" % volname,
+            "kube_hostname": storage.get("node", ""),
+            "node": get_brick_hostname(volname, idx),
+            "node_id": storage["node_id"],
+            "host_brick_path": storage.get("path", ""),
+            "brick_device": storage.get("device", ""),
+            "pvc_name": storage.get("pvc", ""),
+            "brick_device_dir": get_brick_device_dir(storage),
+            "brick_index": idx
+        })
+
+    volinfo_file = "%s.info" % volname
+    configmap_data.data[volinfo_file] = ""
+
+    core_v1_client.patch_namespaced_config_map(
+        KADALU_CONFIG_MAP, NAMESPACE, configmap_data)
+    logging.info(logf("Updated configmap", name=KADALU_CONFIG_MAP,
+                      volname=volname))
+
+
+def delete_server_pods(obj):
+    """
+    Delete server pods depending on type of Hosting
+    Volume and other options specified
+    """
+    # Delete server pod
+    volname = obj["metadata"]["name"]
+    voltype = obj["spec"]["type"]
+    docker_user = os.environ.get("DOCKER_USER", "kadalu")
+
+    shd_required = False
+    if voltype in (VOLUME_TYPE_REPLICA_3, VOLUME_TYPE_REPLICA_2):
+        shd_required = True
+
+    template_args = {
+        "namespace": NAMESPACE,
+        "kadalu_version": VERSION,
+        "docker_user": docker_user,
+        "volname": volname,
+        "voltype": voltype,
+        "volume_id": obj["spec"]["volume_id"],
+        "shd_required": shd_required
+    }
+
+    # One StatefulSet per Brick
+    for idx, storage in enumerate(obj["spec"]["storage"]):
+        template_args["host_brick_path"] = storage.get("path", "")
+        template_args["kube_hostname"] = storage.get("node", "")
+        # TODO: Understand the need, and usage of suffix
+        template_args["serverpod_name"] = get_brick_hostname(
+            volname,
+            idx,
+            suffix=False
+        )
+        template_args["brick_path"] = "/bricks/%s/data/brick" % volname
+        template_args["brick_index"] = idx
+        template_args["brick_device"] = storage.get("device", "")
+        template_args["pvc_name"] = storage.get("pvc", "")
+        template_args["brick_device_dir"] = get_brick_device_dir(storage)
+        template_args["brick_node_id"] = storage["node_id"]
+        template_args["k8s_dist"] = K8S_DIST
+
+        filename = os.path.join(MANIFESTS_DIR, "server.yaml")
+        template(filename, **template_args)
+        lib_execute(KUBECTL_CMD, DELETE_CMD, "-f", filename)
+        logging.info(logf("Deleted Server pod",
+                          volname=volname,
+                          manifest=filename,
+                          node=storage.get("node", "")))
+
+def get_storages():
+    """Parse list of Storages from JSON input"""
+
+    storage = storage_list.Storage()
```

Comment:
Will 'storage' be a class object with default values set or not? 'storage' is already being instantiated inside list_storages() but its not returning 'storage' or 'storage_unit' but only 'storages'